src/components.jl

```diff
@@ -178,10 +178,7 @@ function get_component(
     name::AbstractString,
 )::Union{T, Nothing} where {T <: InfrastructureSystemsType}
     if !isconcretetype(T)
-        components = get_components(T, components, x -> get_name(x) == name)
-        if length(components) > 1
-            throw(ArgumentError("More than one components of type $T"))
-        end
+        components = get_components_by_name(T, components, name)
```

Comment:
The docstring is now incorrect.

More importantly, this could cause very confusing behaviors for our users.  If they call `get_component(Component)` and there are duplicate names across the concrete types, this will return the first one, which will be determined seemingly randomly be the way `get_components` iterates over component types.  Users will not be able to predict what they will get back.

With the current approach, if the user doesn't know the concrete type then they have to call `get_components_by_name` and then check how many components they received.  That might be a bit cumbersome, but it's easy to understand and allows us to convey clear, consistent rules.

I think we should have a group discussion before merging this functionality.

Reply:
Concrete types don't allow for duplicate names. I understand the concern, but with the current approach, it is way too cumbersome to make models based on names when mixes of types are involved. For instance, setting a reserve value where the contributing devices can have multiple types. 

The objective is to make a simple function that can integrate well with a JuMP macro to define a variable/constraint. Having to do this outside is adding extra overhead for a corner case that would fail anyway if the user has repeated names in the definition of the set.